src/libraries/System.Runtime/tests/System/ObsoleteAttributeTests.cs

```diff
@@ -14,28 +17,69 @@ public static void Ctor_Default()
             var attribute = new ObsoleteAttribute();
             Assert.Null(attribute.Message);
             Assert.False(attribute.IsError);
+            Assert.Null(attribute.DiagnosticId);
+            Assert.Null(attribute.UrlFormat);
         }
 
         [Theory]
-        [InlineData(null)]
-        [InlineData("")]
-        [InlineData("message")]
-        public void Ctor_String(string message)
+        [InlineData(null, null)]
+        [InlineData("", "BCL0006")]
+        [InlineData("message", "")]
+        public void Ctor_String_Id(string message, string id)
         {
-            var attribute = new ObsoleteAttribute(message);
+            var attribute = new ObsoleteAttribute(message) { DiagnosticId = id};
             Assert.Equal(message, attribute.Message);
             Assert.False(attribute.IsError);
+            Assert.Equal(id, attribute.DiagnosticId);
+            Assert.Null(attribute.UrlFormat);
         }
 
         [Theory]
-        [InlineData(null, true)]
-        [InlineData("", false)]
-        [InlineData("message", true)]
-        public void Ctor_String_Bool(string message, bool error)
+        [InlineData(null, true, "")]
+        [InlineData("", false, null)]
+        [InlineData("message", true, "https://aka.ms/obsolete/{0}")]
+        public void Ctor_String_Bool_Url(string message, bool error, string url)
         {
-            var attribute = new ObsoleteAttribute(message, error);
+            var attribute = new ObsoleteAttribute(message, error) { UrlFormat = url};
             Assert.Equal(message, attribute.Message);
             Assert.Equal(error, attribute.IsError);
+            Assert.Null(attribute.DiagnosticId);
+            Assert.Equal(url, attribute.UrlFormat);
         }
+
+        [Theory]
+        [MemberData(nameof(TestData_Attributes))]
+        public void TestAttributeValuesWithReflection(Type testType, string message, bool error, string id, string url) {
+
+            Attribute attribute = Attribute.GetCustomAttributes(testType, typeof(Attribute)).Where((e) => e.GetType() == typeof(ObsoleteAttribute)).SingleOrDefault();
```

Comment:
Does this test add anything? ie., assuming hte other tests pass, what bug could it find in the type?

Reply:
Well this looks for me like testing an attribute closer to real use, maybe i use it with different targets not only class, then it might add little more sense

Reply:
This is testing reflection APIs. It is not testing the attribute. We do not have tests like this for any other attributes.

Reply:
I thought i am using reflection API for testing 😛, from other side you are right, well will remove them